download_test_vectors.sh

```diff
@@ -30,28 +33,72 @@ cleanup() {
 trap cleanup SIGINT
 
 dl_version() {
-	[[ -z "$1" ]] && { echo "usage: dl_version() vX.Y.Z"; exit 1; }
+	[[ -z "$1" ]] && { echo "usage: dl_version() vX.Y.Z | nightly"; exit 1; }
 	version="$1"
 
 	mkdir -p "tarballs/${version}"
 	pushd "tarballs/${version}" >/dev/null
-	for flavour in "${FLAVOURS[@]}"; do
-		if [[ ! -e "${flavour}.tar.gz" ]]; then
-			echo "Downloading: ${version}/${flavour}.tar.gz"
-			curl --location --remote-name --silent --show-error --retry 3 --retry-all-errors \
-				"https://github.com/ethereum/consensus-spec-tests/releases/download/${version}/${flavour}.tar.gz" \
-				|| {
-					echo "Curl failed. Aborting"
-					rm -f "${flavour}.tar.gz"
+
+	if [[ "$version" == "nightly" ]]; then
+		if [[ -z "${GITHUB_TOKEN:-}" ]]; then
+			echo "Error GITHUB_TOKEN is not set"
+			exit 1
+		fi
+
+		repo="ethereum/consensus-specs"
+		api="https://api.github.com"
+		auth_header="Authorization: token ${GITHUB_TOKEN}"
+
+		run_id=$(curl -s -H "${auth_header}" \
+			"${api}/repos/${repo}/actions/workflows/generate_vectors.yml/runs?branch=dev&status=success&per_page=1" \
+			| jq -r '.workflow_runs[0].id')
+
+		if [[ "${run_id}" == "null" || -z "${run_id}" ]]; then
+			echo "No successful nightly workflow run found"
+			exit 1
+		fi
+
+		echo "Downloading nightly test vectors for run: ${run_id}"
+		curl -s -H "${auth_header}" "${api}/repos/${repo}/actions/runs/${run_id}/artifacts" |
+			jq -c '.artifacts[] | {name, url: .archive_download_url}' |
+			while read -r artifact; do
+				name=$(echo "${artifact}" | jq -r .name)
+				url=$(echo "${artifact}" | jq -r .url)
+
+				if [[ "$name" == "consensustestgen.log" ]]; then
+					continue
+				fi
+
+				echo "Downloading artifact: ${name}"
+				curl --progress-bar --location --show-error --retry 3 --retry-all-errors \
+						-H "${auth_header}" -H "Accept: application/vnd.github+json" \
+						--output "${name}.zip" "${url}" || {
+					echo "Failed to download ${name}"
 					exit 1
 				}
-		fi
-	done
+
+				unzip -qo "${name}.zip"
```

Comment:
this is a new runtime dependency, potentially; I'm not sure if it's in all the images we use (which includes Windows, macOS, and Ubuntu Linux 22.04 or newer)